walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,94 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
 
 public final class FindMeetingQuery {
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    List<TimeRange> possibleTimes = new ArrayList<TimeRange>();
+    Set<String> attendees =  new HashSet<String>();
+    int meetingDur = (int) request.getDuration();
+    attendees.addAll(request.getAttendees());
+    if (meetingDur > TimeRange.WHOLE_DAY.duration()){
+      return possibleTimes;
+    }
+    if (events.isEmpty() || attendees.isEmpty()){
+      possibleTimes.add(TimeRange.WHOLE_DAY);
+      return possibleTimes;
+    }
+    ArrayList<TimeRange> busyWithOverlap = getBusyTimes(events, attendees);
+    possibleTimes = findFreeTimes(busyWithOverlap, request.getDuration());
+    return possibleTimes;
   }
-}
+
+/**
+  * Returns an ArrayList of meeting times that are unavailable due to schedule
+  * conflicts and overlapping meeting times within the given group of attendees. 
+  *
+  * @param  events the Collection of Event objects passed into the query function
```

Comment:
minor nit.
These parameters don't actually have to be from the query. This function works to get the busy times given any collection of events and set of attendees.  You are only using it for this query, but there is no need to specify that in the comments here that those params must come from the query

Reply:
True, updated the javadoc to reflect this!